config/keys.go

```diff
@@ -65,5 +71,27 @@ func LoadSigningKeys() (*SigningKeys, error) {
 		}
 		return nil, err
 	}
+
+	if err := validateKeys(&config); err != nil {
+		return nil, err
+	}
+
 	return &config, nil
 }
+
+func validateKeys(config *SigningKeys) error {
+	keys := config.Keys
+	uniqMap := make(map[string]struct{}, len(keys))
+	for _, key := range keys {
+		if _, ok := uniqMap[key.Name]; ok {
+			return fmt.Errorf("malformed %s: multiple keys with name '%s' found", dir.PathSigningKeys, key.Name)
+		}
+		uniqMap[key.Name] = struct{}{}
+	}
+
+	if _, ok := uniqMap[config.Default]; !ok {
```

Comment:
If default key is not specified, it is also an valid signingkeys.json file?

Reply:
That's correct, default key is optional.